pysat/tests/test_instrument.py

```diff
@@ -29,25 +29,27 @@ def setup(self):
                                          sat_id='10',
                                          clean_level='clean',
                                          update_files=True)
+        self.ref_time = dt.datetime(2009, 1, 1)
+        self.ref_doy = 1
+        self.out = None
 
     def teardown(self):
         """Runs after every method to clean up previous testing."""
-        del self.testInst
+        del self.testInst, self.out, self.ref_time, self.ref_doy
 
-    # --------------------------------------------------------------------------
+    # -------------------------------------------------------------------------
     #
     # Test basic loads, by date, filename, file id, as well as prev/next
     #
-    # --------------------------------------------------------------------------
+    # -------------------------------------------------------------------------
     def test_basic_instrument_load(self):
         """Test if the correct day is being loaded (checking object date and
         data)."""
-        self.testInst.load(2009, 1)
-        test_date = self.testInst.index[0]
-        test_date = dt.datetime(test_date.year, test_date.month,
-                                test_date.day)
-        assert (test_date == dt.datetime(2009, 1, 1))
-        assert (test_date == self.testInst.date)
+        self.testInst.load(self.ref_time.year, self.ref_doy)
+        self.out = self.testInst.index[0]
```

Comment:
Missed all these changes the first time :P

Can you provide some background on why internal variables within the test functions are being assigned to the test object?

Reply:
No, you just missed the commit history :P 

This is reducing redundancy in the test code and improving its maintainability.  When you do this, you also can clearly see how many of the tests are duplicates of each other and require parametrization, but I opened an issue for that already.